core/Archive/ArchiveInvalidator.php

```diff
@@ -413,12 +429,54 @@ public function markArchivesOverlappingRangeAsInvalidated(array $idSites, array
         return $invalidationInfo;
     }
 
+    /**
+     * Schedule rearchiving of reports for a single plugin or single report for N months in the past. The next time
+     * core:archive is run, they will be processed.
+     * TODO test
+     * @param int[] $idSite
+     * @param Date $date1
+     * @param Date $date2
+     * @param string $plugin
+     * @param string|null $report
+     * @throws \Exception
+     * @api
+     */
+    public function reArchiveReport(array $idSites, string $plugin, string $report = null, int $lastNMonthsToInvalidate = null)
+    {
+        $lastNMonthsToInvalidate = $lastNMonthsToInvalidate ?: Config::getInstance()->General['rearchive_reports_in_past_last_n_months'];
+        if (empty($lastNMonthsToInvalidate)) {
+            return;
+        }
+
+        $lastNMonthsToInvalidate = (int) substr($lastNMonthsToInvalidate, 4);
```

Comment:
btw not sure if `markArchivesAsInvalidated ` does it maybe but if the site was created only 60 days ago, and configured is to archive 6 months back, then we would only invalidate the last 60days.

Reply:
👍